### ISSUE TITLE
Error cannot fetch your preferences when not logged in

### ISSUE THREAD

**Issue by bourgeoa** [solid/solid-ui#464]
```
Unable to fetch your preferences - this was the error:  Error: (via loadPrefs) Error: Can't log in: Error: Could not log in
    at _callee5$ (authn.ts:288)
    at tryCatch (runtime.js:63)
    at Generator.invoke [as _invoke] (runtime.js:294)
    at Generator.throw (runtime.js:119)
    at asyncGeneratorStep (asyncToGenerator.js:3)
    at _throw (asyncToGenerator.js:29)
```

The error is raised in https://github.com/solid/solid-ui/blob/6a9e81a980072cb7f7b8e8098346bd1acac55269/src/authn/authn.ts#L219

**Comment by SharonStrats** [comment]
Hi @bourgeoa I'm wondering what would the expected behavior be? Should we just be told that we are not logged in?

**Comment by bourgeoa** [comment]
> what would the expected behavior be? Should we just be told that we are not logged in?

@SharonStrats Profile and public preferences should be available.

> I also can't seem to recreate it. I'm wondering if you could show the steps you took to produce it. I logged in, went to preferences, logged out and it just refreshes and goes to my public data.

Logging out do not clear public preferences

**Comment by ewingson** [comment]
@bourgeoa @SharonStrats 

maybe this is a good first issue to wrap my head around.
I' m trying to reproduce it. being *not* logged in and opening the `settings` container throws a 401 (see pic).

what is the desired behaviour ? being in which state and doing what exact steps shows the error ?
where would the public preferences be meant to be shown ?

![Bildschirmfoto_2022-01-09_11-31-32](https://user-images.githubusercontent.com/36561948/148678643-5655aba4-ee2e-4ed3-9b3f-2f1c26eff285.png)

**Comment by bourgeoa** [comment]
@ewingson 
The error appears in the browser  developper : 
![image](https://user-images.githubusercontent.com/2218470/148680892-d1e89043-59d6-4df6-82a1-899f4e95a12c.png)

**Comment by ewingson** [comment]
@bourgeoa @SharonStrats 

I was able to reproduce it (see pic).
Alain, you already tracked it down fortunately.
Dunno though if the code seems to trigger (or need) a Login.

maybe we must remove the triggering mechanism `solidLogicSingelton`.

**Comment by ewingson** [comment]
@bourgeoa @SharonStrats 

I tried to follow the code flow.
to me the console says Line 288. ((see two pics)).
![Bildschirmfoto_2022-01-09_14-07-58](https://user-images.githubusercontent.com/36561948/148683454-1ae7026d-fc7a-447a-b0c0-750b19bfe1bd.png)
![Bildschirmfoto_2022-01-09_14-07-12](https://user-images.githubusercontent.com/36561948/148683465-09a5121f-6d73-4bbc-b804-4d9e7d1e5f06.png)

**Comment by timea-solid** [comment]
Since we are not logged in it should not even try to read the preference file. The check for the login should happen before in the code. @jeff-zucker am I correct to say that the app should not even look for it?

**Comment by jeff-zucker** [comment]
Yes, I believe that is correct, the preferencesFile is private and can only be accessed when logged in so there is no point in looking for it if not logged in.  I believe the correct behavior is 
1. Load the WebID Profile Document
2. If logged in, load the preferences file
  2a.  If no preferences file found, offer to create one
3. If not logged in, perhaps informative message in console :"not  logged in, no preferences loaded"

**Comment by timea-solid** [comment]
This was also discussed in SolidOS [team meeting](https://solidos.solidcommunity.net/public/SolidOS%20team%20meetings/2022/2022_04_27.md) (see under tech topics). 
Same steps as Jeff mentioned a bit more implementation detail.

**Comment by SharonStrats** [comment]
I believe this was closed with post-milestone3m solid-logic and solid-ui.
